python/paddle/fluid/tests/unittests/test_one_hot_op.py

```diff
@@ -51,19 +52,20 @@ class TestOneHotOp_default_dtype(OpTest):
     def setUp(self):
         self.op_type = 'one_hot'
         depth = 10
+        depth_np = np.array(10).astype('int32')
         dimension = 12
         x_lod = [[4, 1, 3, 3]]
         x = [np.random.randint(0, depth - 1) for i in range(sum(x_lod[0]))]
-        x = np.array(x).astype('int').reshape([sum(x_lod[0]), 1])
+        x = np.array(x).astype('int32').reshape([sum(x_lod[0]), 1])
 
         out = np.zeros(shape=(np.product(x.shape[:-1]),
                               depth)).astype('float32')
 
         for i in range(np.product(x.shape)):
             out[i, x[i]] = 1.0
 
-        self.inputs = {'X': (x, x_lod)}
-        self.attrs = {'depth': depth}
+        self.inputs = {'X': (x, x_lod), 'depth_tensor': depth_np}
+        self.attrs = {}
```

Comment:
This is not the situation that we should only make the depth as the input but not attribute.